### PR TITLE
Add Zahlenraten

### PR DESCRIPTION
**Pull request by MarcRaeder**

### CODE REVIEW

**Comment by stefanluth** [comment]
`make run` returns this for me:

image

**Comment by stefanluth** [comment]
image

Unnecessarily long data output associations (arrows) :) Also, try to use right angles :P

File: Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlenratenCustomForm.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  constructor(props: CustomFormProps) {
+    super(props);
+    this.userInput = React.createRef();
+  }  
+
+  public render(): JSX.Element {
+    return (
+      <div className='test-class'>
+
+        <h1>Du hast noch {this.props.userTask.tokens[0].payload.leftTries} Versuche</h1>
```
Comment (stefanluth) [inline Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlenratenCustomForm.tsx:21]:
```suggestion
      <div className='test-class'>
        <h1>Du hast noch {this.props.userTask.tokens[0].payload.leftTries} Versuche</h1>
```

Comment (stefanluth) [reply Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlenratenCustomForm.tsx:21]: You use `this.props.userTask.tokens[0].payload` a lot, you could move it to a variable :)

File: Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlenratenCustomForm.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import { CustomFormProps } from '../DialogRenderer';
+
+import './ZahlenratenCustomForm.css';
+
+export class ZahlenratenCustomForm extends React.Component<CustomFormProps> {
+
+  private userInput: React.RefObject<HTMLInputElement>;
```
Comment (stefanluth) [inline Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlenratenCustomForm.tsx]:
```suggestion
export class ZahlenratenCustomForm extends React.Component<CustomFormProps> {
  private userInput: React.RefObject<HTMLInputElement>;
```

File: Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlZuNiedrigCustomForm.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+import { CustomFormProps } from '../DialogRenderer';
+
+import './ZahlenratenCustomForm.css';
+
+export class ZahlZuNiedrigCustomForm extends React.Component<CustomFormProps> {
+
+  private guess;
```
Comment (stefanluth) [inline Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlZuNiedrigCustomForm.tsx]:
```suggestion
export class ZahlZuNiedrigCustomForm extends React.Component<CustomFormProps> {
  private guess;
```

File: Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlZuHochCustomForm.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  private guess;
+  private randomNumber;
+  private text;
+  private range;
+  private tries;
+  private distance;
+  
+
+  constructor(props: CustomFormProps) {
```
Comment (stefanluth) [inline Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlZuHochCustomForm.tsx]:
```suggestion
export class ZahlZuHochCustomForm extends React.Component<CustomFormProps> {
  private guess;
  private randomNumber;
  private text;
  private range;
  private tries;
  private distance;

  constructor(props: CustomFormProps) {
```

File: Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlZuNiedrigCustomForm.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  private randomNumber;
+  private text;
+  private range;
+  private tries;
+  private distance;
+  
+
+  constructor(props: CustomFormProps) {
```
Comment (stefanluth) [inline Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenCustomForm/ZahlZuNiedrigCustomForm.tsx]:
```suggestion
  private distance;

  constructor(props: CustomFormProps) {
```

File: Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenStartDialog/ZahlenratenStartdialog.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { StartDialogProps } from '../DialogRenderer';
+
+export class ZahlenratenStartDialog extends React.Component<StartDialogProps> {
+
+  private userName: React.RefObject<HTMLInputElement>;
+  private lowestNumber: React.RefObject<HTMLInputElement>;
+  private highestNumber: React.RefObject<HTMLInputElement>;
+
+
+  constructor(props: StartDialogProps) {
```
Comment (stefanluth) [inline Zahlenraten.Intermediate/apps/Zahlenraten.Intermediate/frontend/src/ZahlenratenStartDialog/ZahlenratenStartdialog.tsx]:
```suggestion
export class ZahlenratenStartDialog extends React.Component<StartDialogProps> {
  private userName: React.RefObject<HTMLInputElement>;
  private lowestNumber: React.RefObject<HTMLInputElement>;
  private highestNumber: React.RefObject<HTMLInputElement>;

  constructor(props: StartDialogProps) {
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

**Review by stefanluth — APPROVED** [review]

**Review by DevBastiS — CHANGES_REQUESTED** [review]
build folder in frontend should be ignored.
Otherwise this looks really good.

**Review by DevBastiS — APPROVED** [review]

**Pull request merged**